### PR TITLE
feat: Gmail Loader component created

### PR DESCRIPTION
**Pull request by edwinjosechittilappilly**
Added Custom Gmail Chat loader to load emails using service account token as json string

Requesting review:
@jordanrfrazier  @nicoloboschi

### CODE REVIEW

File: src/backend/base/langflow/components/data/Gmail.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import json
+from typing import Any, Iterator, List
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from langflow.custom import Component
+from langflow.inputs import MessageTextInput
+from langflow.io import SecretStrInput
+from langflow.template import Output
+from langflow.schema import Data
+from langchain_community.chat_loaders.gmail import GMailLoader
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/components/data/Gmail.py:12]: It seems this is deprecated. Could you take a look at https://github.com/langchain-ai/langchain-google/tree/main/libs/community/langchain_google_community

File: src/backend/base/langflow/components/data/Gmail.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            value="10",
+        ),
+    ]
+
+    outputs = [
+        Output(display_name="Loaded Emails", name="emails", method="load_emails"),
+    ]
+
+    def load_emails(self) -> Data:
+        class CustomGMailLoader(GMailLoader):
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/components/data/Gmail.py:65]: Why do you need to create a custom class from it?

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey @edwinjosechittilappilly 

How are you? 

Thanks for this component. It is looking good. I've left some comments.

File: src/backend/base/langflow/components/data/Gmail.py (hunk 1 above)
Comment (edwinjosechittilappilly) [reply src/backend/base/langflow/components/data/Gmail.py:12]: Sure; I did nottice it. 
Updating it to use Gmail Loaders from langchain_google_community

File: src/backend/base/langflow/components/data/Gmail.py (hunk 2 above)
Comment (edwinjosechittilappilly) [reply src/backend/base/langflow/components/data/Gmail.py:65]: The Class Gmail loader support only for Label ID "SENT" from emails.

I tweaked the support functions and added cleaning functions.
Now, it support the labels "SENT", "INBOX","UNREAD" and "IMPORTANT". Also the message contnet is cleaned (Extra chareacters removed) using the cleaning function.

**Comment by edwinjosechittilappilly** [comment]
Hi @ogabrielluiz 
Thanks for the review. 
I have resolved with comments.

We are using this component as well as the Google Drive component for our internal project as custom components.

Hence thought of contributing these components back to the community. 
Looking forward to show the demo of the browser extension once it completed.

**Review by ogabrielluiz — APPROVED** [review]
Awesome! 

LGTM

**Pull request merged**